hls4ml/writer/vivado_writer.py

```diff
@@ -658,7 +660,14 @@ def keras_model_representer(dumper, keras_model):
             pass
 
         with open(model.config.get_output_dir() + '/' + config_filename, 'w') as file:
-            yaml.dump(model.config.config, file)
+            try:
+                yaml.dump(model.config.config, file)
+            except:
+                import torch
+                model_path = model.config.get_output_dir() + "/torch_model_state_dict.pt"
+                torch.save(model.config.config["PytorchModel"].state_dict(), model_path)
+                model.config.config["PytorchModel"] = model_path
+                yaml.dump(model.config.config, file)
```

Comment:
I don't think this is safe. If the previous `try` failed, it doesn't mean the model is a `torch` model, in the general case.

Reply:
Now it will skip writing the hls4ml_config.yml file at all, and raise a UserWarning instead: https://github.com/abdelabd/hls4ml/commit/7e7ebf18401fac47714151b3d9d9527bdf934e13

As far as I can tell, the file isn't necessary for creating, compiling, or building the HLSModel

Reply:
The yaml file is very useful for reproducibility and it also allows you to split your conversion into several steps, that you may run at different times on different computers, so I'd say we shouldn't remove the feature. I fully agree that the current way of writing this information in yaml is not the best (the Keras one also has problems), but let's improve on that instead of dropping the feature.

Reply:
Okay, another easy fix would be to save the YAML file, but to first delete the "PytorchModel" argument from the `model.config.config`. I'm trying to avoid messing with the `yaml.dump` method, but if this fix still seems too messy then I guess that can't be avoided.

Reply:
Like this: https://github.com/abdelabd/hls4ml/commit/bc93393300a233f11ebb749b0e6bb0a407d852fb